File: Discord_android/Discord/app/src/main/java/com/daophilac/discord/Inventory.java
package com.daophilac.discord;

import com.daophilac.discord.models.Channel;
import com.daophilac.discord.models.Message;
import com.daophilac.discord.models.Server;
import com.daophilac.discord.models.User;
import com.daophilac.discord.tools.JSONConverter;

import java.util.List;

public class Inventory {
    private JSONConverter jsonConverter = new JSONConverter();
    private Channel currentChannel;
    private User currentUser;
    private List<Server> listServer;
    private List<Channel> listChannel;
    private List<Message> listMessage;
    public void storeCurrentChannel(Channel channel){
        this.currentChannel = channel;
    }
    public void storeCurrentChannel(String json){
        this.currentChannel = jsonConverter.toChannel(json);
    }
    public Channel loadCurrentChannel(){
        return this.currentChannel;
    }
    public void storeCurrentUser(User user){
        this.currentUser = user;
    }
    public void storeCurrentUser(String json){
        this.currentUser = jsonConverter.toUser(json);
    }
    public User loadCurrentUser(){
        return this.currentUser;
    }
    public void storeListServer(List<Server> listServer){
        this.listServer = listServer;
    }
    public void storeListServer(String json){
        this.listServer = this.jsonConverter.toListServer(json);
    }
    public List<Server> loadListServer(){
        return this.listServer;
    }
    public void storeListChannel(List<Channel> listChannel){
        this.listChannel = listChannel;
    }
    public void storeListChannel(String json){
        this.listChannel = this.jsonConverter.toListChannel(json);
    }
    public List<Channel> loadListChannel(){
        return this.listChannel;
    }
    public void storeListMessage(List<Message> listMessage){
        this.listMessage = listMessage;
    }
    public void storeListMessage(String json){
        this.listMessage = this.jsonConverter.toListMessage(json);
    }
    public List<Message> loadListMessage(){
        return this.listMessage;
    }
}

File: Discord_win/Discord_win/Inventory.cs
using Discord_win.Models;
using Discord_win.Tools;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Discord_win {
    public class Inventory {
        //private JSONConverter jsonConverter;
        private Channel currentChannel;
        private User currentUser;
        private List<Server> listServer;
        private List<Channel> listChannel;
        private List<Message> listMessage;
        public Inventory() {
            //this.jsonConverter = new JSONConverter();
        }
        public void StoreCurrentChannel(Channel channel) {
            this.currentChannel = channel;
        }
        public void StoreCurrentChannel(string json) {
            //this.currentChannel = jsonConverter.ToChannel(json);
            this.currentChannel = JsonConvert.DeserializeObject<Channel>(json);
        }
        public Channel LoadCurrentChannel() {
            return this.currentChannel;
        }
        public void StoreCurrentUser(User user) {
            this.currentUser = user;
        }
        public void StoreCurrentUser(string json) {
            //this.currentUser = jsonConverter.ToUser(json);
            this.currentUser = JsonConvert.DeserializeObject<User>(json);
        }
        public User LoadCurrentUser() {
            return this.currentUser;
        }
        public void StoreListServer(List<Server> listServer) {
            this.listServer = listServer;
        }
        public void StoreListServer(string json) {
            //this.listServer = this.jsonConverter.ToListServer(json);
            this.listServer = JsonConvert.DeserializeObject<List<Server>>(json);
        }
        public List<Server> LoadListServer() {
            return this.listServer;
        }
        public void StoreListChannel(List<Channel> listChannel) {
            this.listChannel = listChannel;
        }
        public void StoreListChannel(string json) {
            //this.listChannel = this.jsonConverter.ToListChannel(json);
            this.listChannel = JsonConvert.DeserializeObject<List<Channel>>(json);
        }
        public List<Channel> LoadListChannel() {
            return this.listChannel;
        }
        public void StoreListMessage(List<Message> listMessage) {
            this.listMessage = listMessage;
        }
        public void StoreListMessage(string json) {
            //this.listMessage = this.jsonConverter.ToListMessage(json);
            this.listMessage = JsonConvert.DeserializeObject<List<Message>>(json);
        }
        public List<Message> LoadListMessage() {
            return this.listMessage;
        }
    }
}
